packages/ruleset/src/functions/valid-type-format.js

```diff
@@ -0,0 +1,118 @@
+const { validateSubschemas } = require('../utils');
+
+// Valid schema types.
+const validTypes = [
```

Comment:
Technically, for a swagger 2 document, this would also include 'file' as a valid type but we have a separate issue open to address the oas2 rule related to the 'file' type.